service/controller/appcatalog/v1/resource/index/current_test.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/giantswarm/micrologger/microloggertest"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_Resource_GetCurrentState(t *testing.T) {
+	tests := []struct {
+		name              string
+		obj               *v1alpha1.AppCatalog
+		returnedConfigMap *corev1.ConfigMap
+		errorMatcher      func(error) bool
+	}{
+		{
+			name: "case 0: index configMap already created",
+			obj: &v1alpha1.AppCatalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "giantswarm",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app-operator.giantswarm.io/version": "1.0.0",
+					},
+				},
+				Spec: v1alpha1.AppCatalogSpec{
+					Title:       "Giant Swarm",
+					Description: "Catalog of Apps by Giant Swarm",
+					Storage: v1alpha1.AppCatalogSpecStorage{
+						Type: "helm",
+						URL:  "https://giantswarm.github.com/app-catalog/",
+					},
+					LogoURL: "https://s.giantswarm.io/...",
+				},
+			},
+			returnedConfigMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "giantswarm-index",
+					Namespace: "giantswarm",
+					Labels: map[string]string{
+						"giantswarm.io/managed-by": "app-operator",
+					},
+				},
+				Data: map[string]string{
+					"index.yaml": "test yaml",
+				},
+			},
+		},
+		{
+			name: "case 1: index configMap not found",
+			obj: &v1alpha1.AppCatalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "giantswarm",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app-operator.giantswarm.io/version": "1.0.0",
+					},
+				},
+				Spec: v1alpha1.AppCatalogSpec{
+					Title:       "Giant Swarm",
+					Description: "Catalog of Apps by Giant Swarm",
+					Storage: v1alpha1.AppCatalogSpecStorage{
+						Type: "helm",
+						URL:  "https://giantswarm.github.com/app-catalog/",
+					},
+					LogoURL: "https://s.giantswarm.io/...",
+				},
+			},
+			returnedConfigMap: nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			objs := make([]runtime.Object, 0, 0)
+			if tc.returnedConfigMap != nil {
+				objs = append(objs, tc.returnedConfigMap)
+			}
+
+			k8sClient := fake.NewSimpleClientset(objs...)
+
+			c := Config{
+				K8sClient: k8sClient,
+				Logger:    microloggertest.New(),
+
+				IndexNamespace: "giantswarm",
+				ProjectName:    "app-operator",
+			}
+			r, err := New(c)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			result, err := r.GetCurrentState(context.Background(), tc.obj)
+			switch {
+			case err != nil && tc.errorMatcher == nil:
+				t.Fatalf("error == %#v, want nil", err)
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case err != nil && !tc.errorMatcher(err):
+				t.Fatalf("error == %#v, want matching", err)
+			}
+
+			if err == nil && tc.errorMatcher == nil {
```

Comment:
IMHO I think this condition is not needed, it is obvious that we run into this case if the test didn't failed at this point.